From 43542cc07ee110abe25a1e176238fd6d921d593b Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Mon, 9 Aug 2021 23:07:12 +0200
Subject: Missing include

(for std::unique_ptr, with recent libstdc++ 12 trunk)

Change-Id: I61b7823dd740ea7cdfe0d7403a50ac73b24d1c4e
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/120229
Tested-by: Jenkins
Reviewed-by: Stephan Bergmann <sbergman@redhat.com>
---
 chart2/source/view/inc/PlottingPositionHelper.hxx                     | 4 ++++
 compilerplugins/clang/test/unusedfields.cxx                           | 1 +
 connectivity/source/inc/java/tools.hxx                                | 4 ++++
 filter/source/xsltdialog/xmlfilterjar.hxx                             | 1 +
 forms/source/component/EventThread.hxx                                | 1 +
 forms/source/xforms/propertysetbase.hxx                               | 1 +
 fpicker/source/office/contentenumeration.hxx                          | 4 ++++
 include/oox/crypto/StrongEncryptionDataSpace.hxx                      | 4 ++++
 include/sfx2/sidebar/SidebarPanelBase.hxx                             | 4 ++++
 include/svtools/imageresourceaccess.hxx                               | 4 ++++
 include/vcl/BitmapBuffer.hxx                                          | 4 ++++
 include/vcl/filter/PngImageReader.hxx                                 | 4 ++++
 shell/source/backends/kf5be/kf5backend.cxx                            | 2 ++
 sot/source/unoolestorage/xolesimplestorage.hxx                        | 2 ++
 starmath/inc/token.hxx                                                | 1 +
 .../unodialogs/textconversiondlgs/chinese_translation_unodialog.hxx   | 4 ++++
 sw/source/uibase/docvw/OverlayRanges.hxx                              | 1 +
 sw/source/uibase/docvw/ShadowOverlayObject.hxx                        | 4 ++++
 xmlsecurity/source/gpg/SecurityEnvironment.hxx                        | 3 +++
 xmlsecurity/source/helper/ooxmlsecparser.hxx                          | 1 +
 xmlsecurity/source/helper/xsecparser.hxx                              | 1 +
 21 files changed, 55 insertions(+)

diff --git a/chart2/source/view/inc/PlottingPositionHelper.hxx b/chart2/source/view/inc/PlottingPositionHelper.hxx
index 0962868b0f31..d687cb902453 100644
--- a/chart2/source/view/inc/PlottingPositionHelper.hxx
+++ b/chart2/source/view/inc/PlottingPositionHelper.hxx
@@ -18,6 +18,10 @@
  */
 #pragma once
 
+#include <sal/config.h>
+
+#include <memory>
+
 #include <chartview/ExplicitScaleValues.hxx>
 
 #include <basegfx/range/b2drectangle.hxx>
diff --git a/compilerplugins/clang/test/unusedfields.cxx b/compilerplugins/clang/test/unusedfields.cxx
index 5305b4aec964..6ff9189eab27 100644
--- a/compilerplugins/clang/test/unusedfields.cxx
+++ b/compilerplugins/clang/test/unusedfields.cxx
@@ -14,6 +14,7 @@
 // expected-no-diagnostics
 #else
 
+#include <memory>
 #include <vector>
 #include <ostream>
 #include <com/sun/star/uno/Any.hxx>
diff --git a/connectivity/source/inc/java/tools.hxx b/connectivity/source/inc/java/tools.hxx
index bc9759fa9a88..44a35ecee3f9 100644
--- a/connectivity/source/inc/java/tools.hxx
+++ b/connectivity/source/inc/java/tools.hxx
@@ -19,6 +19,10 @@
 
 #pragma once
 
+#include <sal/config.h>
+
+#include <memory>
+
 #include <com/sun/star/beans/PropertyValue.hpp>
 #include <com/sun/star/uno/Sequence.h>
 
diff --git a/filter/source/xsltdialog/xmlfilterjar.hxx b/filter/source/xsltdialog/xmlfilterjar.hxx
index 6f02e5f99de0..1d8e52fc3e18 100644
--- a/filter/source/xsltdialog/xmlfilterjar.hxx
+++ b/filter/source/xsltdialog/xmlfilterjar.hxx
@@ -22,6 +22,7 @@
 #include <com/sun/star/lang/XSingleServiceFactory.hpp>
 #include <com/sun/star/uno/XComponentContext.hpp>
 
+#include <memory>
 #include <vector>
 
 class filter_info_impl;
diff --git a/forms/source/component/EventThread.hxx b/forms/source/component/EventThread.hxx
index be9e9696734e..20aff73113db 100644
--- a/forms/source/component/EventThread.hxx
+++ b/forms/source/component/EventThread.hxx
@@ -21,6 +21,7 @@
 
 #include <sal/config.h>
 
+#include <memory>
 #include <vector>
 
 #include <com/sun/star/lang/XEventListener.hpp>
diff --git a/forms/source/xforms/propertysetbase.hxx b/forms/source/xforms/propertysetbase.hxx
index bb27f8460fea..969253a04d62 100644
--- a/forms/source/xforms/propertysetbase.hxx
+++ b/forms/source/xforms/propertysetbase.hxx
@@ -26,6 +26,7 @@
 #include <com/sun/star/beans/PropertyAttribute.hpp>
 
 #include <map>
+#include <memory>
 
 namespace com::sun::star::uno {
         class Any;
diff --git a/fpicker/source/office/contentenumeration.hxx b/fpicker/source/office/contentenumeration.hxx
index 9dcfae687883..bc2180c53c47 100644
--- a/fpicker/source/office/contentenumeration.hxx
+++ b/fpicker/source/office/contentenumeration.hxx
@@ -19,6 +19,10 @@
 
 #pragma once
 
+#include <sal/config.h>
+
+#include <memory>
+
 #include <com/sun/star/ucb/XCommandEnvironment.hpp>
 #include <salhelper/thread.hxx>
 #include <ucbhelper/content.hxx>
diff --git a/include/oox/crypto/StrongEncryptionDataSpace.hxx b/include/oox/crypto/StrongEncryptionDataSpace.hxx
index 59be5c9832a0..cd7156515ab3 100644
--- a/include/oox/crypto/StrongEncryptionDataSpace.hxx
+++ b/include/oox/crypto/StrongEncryptionDataSpace.hxx
@@ -11,6 +11,10 @@
 #ifndef INCLUDED_OOX_CRYPTO_STRONGENCRYPTINDATASPACE_HXX
 #define INCLUDED_OOX_CRYPTO_STRONGENCRYPTINDATASPACE_HXX
 
+#include <sal/config.h>
+
+#include <memory>
+
 #include <cppuhelper/implbase.hxx>
 #include <com/sun/star/lang/XServiceInfo.hpp>
 #include <com/sun/star/packages/XPackageEncryption.hpp>
diff --git a/include/sfx2/sidebar/SidebarPanelBase.hxx b/include/sfx2/sidebar/SidebarPanelBase.hxx
index 971e286ca74f..8164244e796a 100644
--- a/include/sfx2/sidebar/SidebarPanelBase.hxx
+++ b/include/sfx2/sidebar/SidebarPanelBase.hxx
@@ -18,6 +18,10 @@
  */
 #pragma once
 
+#include <sal/config.h>
+
+#include <memory>
+
 #include <sfx2/dllapi.h>
 
 #include <cppuhelper/compbase.hxx>
diff --git a/include/svtools/imageresourceaccess.hxx b/include/svtools/imageresourceaccess.hxx
index d7e8a5629b11..ccb5b6b69bce 100644
--- a/include/svtools/imageresourceaccess.hxx
+++ b/include/svtools/imageresourceaccess.hxx
@@ -19,6 +19,10 @@
 
 #pragma once
 
+#include <sal/config.h>
+
+#include <memory>
+
 #include <config_options.h>
 #include <svtools/svtdllapi.h>
 
diff --git a/include/vcl/BitmapBuffer.hxx b/include/vcl/BitmapBuffer.hxx
index 79f202cdfa2d..7468cb987f95 100644
--- a/include/vcl/BitmapBuffer.hxx
+++ b/include/vcl/BitmapBuffer.hxx
@@ -20,6 +20,10 @@
 #ifndef INCLUDED_VCL_BITMAPBUFFER_HXX
 #define INCLUDED_VCL_BITMAPBUFFER_HXX
 
+#include <sal/config.h>
+
+#include <memory>
+
 #include <vcl/dllapi.h>
 #include <vcl/BitmapPalette.hxx>
 #include <vcl/ColorMask.hxx>
diff --git a/include/vcl/filter/PngImageReader.hxx b/include/vcl/filter/PngImageReader.hxx
index 2615fd961bd4..bbb5b7c8d672 100644
--- a/include/vcl/filter/PngImageReader.hxx
+++ b/include/vcl/filter/PngImageReader.hxx
@@ -11,6 +11,10 @@
 #ifndef INCLUDED_VCL_FILTER_PNGIMAGEREADER_HXX
 #define INCLUDED_VCL_FILTER_PNGIMAGEREADER_HXX
 
+#include <sal/config.h>
+
+#include <memory>
+
 #include <vcl/dllapi.h>
 
 #include <com/sun/star/uno/Reference.hxx>
diff --git a/shell/source/backends/kf5be/kf5backend.cxx b/shell/source/backends/kf5be/kf5backend.cxx
index 47c829dc5f9b..e448c124d88a 100644
--- a/shell/source/backends/kf5be/kf5backend.cxx
+++ b/shell/source/backends/kf5be/kf5backend.cxx
@@ -19,6 +19,8 @@
 
 #include <sal/config.h>
 
+#include <memory>
+
 #include <QtWidgets/QApplication>
 
 #include <boost/noncopyable.hpp>
diff --git a/sot/source/unoolestorage/xolesimplestorage.hxx b/sot/source/unoolestorage/xolesimplestorage.hxx
index ef0214ba61c3..2ec9f715d873 100644
--- a/sot/source/unoolestorage/xolesimplestorage.hxx
+++ b/sot/source/unoolestorage/xolesimplestorage.hxx
@@ -22,6 +22,8 @@
 
 #include <sal/config.h>
 
+#include <memory>
+
 #include <com/sun/star/embed/XOLESimpleStorage.hpp>
 #include <com/sun/star/lang/XServiceInfo.hpp>
 #include <cppuhelper/implbase.hxx>
diff --git a/starmath/inc/token.hxx b/starmath/inc/token.hxx
index e38b3e5afb5a..808195dca9d3 100644
--- a/starmath/inc/token.hxx
+++ b/starmath/inc/token.hxx
@@ -35,6 +35,7 @@
 
 // std imports
 #include <cassert>
+#include <memory>
 #include <vector>
 #include <stack>
 #include <tuple>
diff --git a/svx/source/unodialogs/textconversiondlgs/chinese_translation_unodialog.hxx b/svx/source/unodialogs/textconversiondlgs/chinese_translation_unodialog.hxx
index 8d2e29f1317a..dbfae4d74697 100644
--- a/svx/source/unodialogs/textconversiondlgs/chinese_translation_unodialog.hxx
+++ b/svx/source/unodialogs/textconversiondlgs/chinese_translation_unodialog.hxx
@@ -19,6 +19,10 @@
 
 #pragma once
 
+#include <sal/config.h>
+
+#include <memory>
+
 #include <com/sun/star/awt/XWindow.hpp>
 #include <com/sun/star/beans/XPropertySet.hpp>
 #include <com/sun/star/lang/XInitialization.hpp>
diff --git a/sw/source/uibase/docvw/OverlayRanges.hxx b/sw/source/uibase/docvw/OverlayRanges.hxx
index e086b8462ea2..829f0c4001dc 100644
--- a/sw/source/uibase/docvw/OverlayRanges.hxx
+++ b/sw/source/uibase/docvw/OverlayRanges.hxx
@@ -22,6 +22,7 @@
 #include <svx/sdr/overlay/overlayobject.hxx>
 #include <basegfx/range/b2drange.hxx>
 
+#include <memory>
 #include <vector>
 
 class SwView;
diff --git a/sw/source/uibase/docvw/ShadowOverlayObject.hxx b/sw/source/uibase/docvw/ShadowOverlayObject.hxx
index df8e27009f5e..cd612f7ce50e 100644
--- a/sw/source/uibase/docvw/ShadowOverlayObject.hxx
+++ b/sw/source/uibase/docvw/ShadowOverlayObject.hxx
@@ -19,6 +19,10 @@
 
 #pragma once
 
+#include <sal/config.h>
+
+#include <memory>
+
 #include <svx/sdr/overlay/overlayobject.hxx>
 
 class SwView;
diff --git a/xmlsecurity/source/gpg/SecurityEnvironment.hxx b/xmlsecurity/source/gpg/SecurityEnvironment.hxx
index 83b4f38b41c8..262578682274 100644
--- a/xmlsecurity/source/gpg/SecurityEnvironment.hxx
+++ b/xmlsecurity/source/gpg/SecurityEnvironment.hxx
@@ -10,6 +10,9 @@
 #pragma once
 
 #include <sal/config.h>
+
+#include <memory>
+
 #include <rtl/ustring.hxx>
 #include <cppuhelper/implbase.hxx>
 
diff --git a/xmlsecurity/source/helper/ooxmlsecparser.hxx b/xmlsecurity/source/helper/ooxmlsecparser.hxx
index 9d0c7efe470d..1ac72a46f0c4 100644
--- a/xmlsecurity/source/helper/ooxmlsecparser.hxx
+++ b/xmlsecurity/source/helper/ooxmlsecparser.hxx
@@ -16,6 +16,7 @@
 
 #include <xmloff/namespacemap.hxx>
 
+#include <memory>
 #include <optional>
 #include <stack>
 
diff --git a/xmlsecurity/source/helper/xsecparser.hxx b/xmlsecurity/source/helper/xsecparser.hxx
index fa2d4a8bc4ab..6279d65439e1 100644
--- a/xmlsecurity/source/helper/xsecparser.hxx
+++ b/xmlsecurity/source/helper/xsecparser.hxx
@@ -26,6 +26,7 @@
 
 #include <xmloff/namespacemap.hxx>
 
+#include <memory>
 #include <optional>
 #include <stack>
 
-- 
cgit v1.2.1

